homeassistant/components/tuneblade_remote/entity.py

```diff
@@ -0,0 +1,45 @@
+"""TuneBladeEntity base class."""
+
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, NAME
+
+
+class TuneBladeEntity(CoordinatorEntity):
+    """Base entity for TuneBlade devices, including master hub."""
+
+    def __init__(
+        self, coordinator, config_entry, device_id=None, device_name=None
+    ) -> None:
+        """Initialize entity with coordinator, config entry, and optional device info."""
+        super().__init__(coordinator)
+        self.config_entry = config_entry
+        self.device_id = device_id or "master"
+        self.device_name = device_name or "Master"
+
+    @property
+    def unique_id(self) -> str | None:
+        """Return a unique ID for this entity."""
+        return f"{self.config_entry.entry_id}_{self.device_id}"
+
+    @property
+    def device_info(self) -> DeviceInfo | None:
+        """Return device info for this entity."""
+        return DeviceInfo(
+            identifiers={(DOMAIN, self.device_id)},
+            name=f"{self.device_name} {NAME}",
+            manufacturer=NAME,
+        )
```

Comment:
Can be moved to the constructor using `_attr_*` (shorthand attributes)

Reply:
OK that's done